### DISCUSSION TITLE
Multiple instances of the same mesh with varying materials

### DISCUSSION THREAD

**Discussion by crs-skilltree** [vpenades/SharpGLTF#254] [Q&A]
I would like to create a mesh (using MeshBuilder) and instantiate this mesh multiple times with varying materials (and transforms). Although the GLTF spec supports this, I did not find a way to do this in SharpGLTF: 
* The only way I found to create an index buffer is to call MeshBuilder.UsePrimitive(..) which requires you to specificy a MaterialBuilder beforehand. How can I reuse the same index buffer to create instances with varying materials?

**Comment by vpenades** [comment]
MeshBuilder and all other classes in the Toolkit library are designed to simplify the authoring of glTF, but there's a tradeoff between ease of use and limited support of rarely used features.

Sadly, your use case is not covered by the toolkit.  If you want to create multiple meshes that reuse the same accessors but have different materials, I would do this:

1- create the 1st mesh using MeshBuilder, create a simple scene and convert to glTF
2- Use the low level Schema2 API to create new meshes, and when doing so, reuse the accessors of the 1st mesh

It's ugly but it's the best thing I can think about right now.
